scalagen-sbt/src/sbt-test/scalagen/simple/build.sbt

```diff
@@ -0,0 +1,79 @@
+import java.nio.file.Paths
+
+import sbt._
+import scala.meta._
+import scala.meta.contrib._
+
+import Generators._
+
+version := "0.1"
+
+scalaVersion := "2.12.4"
+
+generators := Set(
+  Main
+)
+
+TaskKey[Unit]("checkUntouchedHello") := {
+  val relative = Paths.get("main/scala/Hello.scala")
+  val input = IO.read(sourceDirectory.value.toPath.resolve(relative).toFile)
+  val output = IO.read(sourceManaged.value.toPath.resolve(relative).toFile)
+
+  val in = input.parse[Source].get
+  val out = output.parse[Source].get
+
+  if (!(in isEqual out)) {
```

Comment:
I don't know how it works, but in scalafix we can share code between scripted tests with a project template https://github.com/scalacenter/scalafix/tree/master/scalafix-sbt/src/test-project-template we use this feature to have pretty unified diffs on test failures. Debugging equality between two large multiline strings is cumbersome by hand.

Reply:
Okay, I will do that in another patchset